src/app/profile/certificates/services/certificates-service.ts

```diff
@@ -9,15 +9,17 @@ import {
   CreateCertificateDto,
 } from '../models/certificate.model';
 
-const fetchCertificates = async (
-  userId: number
-): Promise<CertificateDto[]> => {
+const fetchCertificates = async (userId: number): Promise<CertificateDto[]> => {
   return get<CertificateDto[]>(`${APIs.CERTIFICATE}?userId=${userId}`);
 };
 
 const createCertificate = async (
   certificate: CreateCertificateDto
 ): Promise<CertificateDto> => {
+  if (certificate.userId) {
+    return post(`${APIs.CERTIFICATE}/create-on-behalf-of`, certificate);
+  }
+
   return post(APIs.CERTIFICATE, certificate);
```

Comment:
What do you think of this approach?

```typescript
const requestUrl = APIs.CERTIFICATE + `${ certificate.userId ? '/create-on-behalf-of' : '' }`;

return post(requestUrl , certificate);
```

Reply:
Works for me! Easier than duplicating methods :D